app/src/main/java/space/taran/arkmemo/files/text/TextNoteFiles.kt

```diff
@@ -0,0 +1,148 @@
+package space.taran.arkmemo.files.text
+
+import android.content.Context
+import android.util.Log
+import space.taran.arklib.computeId
+import space.taran.arkmemo.data.ResourceMeta
+import space.taran.arkmemo.files.parsers.JsonParser
+import space.taran.arkmemo.models.TextNote
+import space.taran.arkmemo.preferences.MemoPreferences
+import java.io.*
+import java.nio.file.Files
+import java.nio.file.Path
+import kotlin.io.path.extension
+import kotlin.io.path.getLastModifiedTime
+
+class TextNoteFiles {
+    companion object {
+        private const val NOTE_EXT = "note"
+        private const val DUMMY_FILENAME =  "Note"
+
+        private fun createTextNoteFile(path: Path?, noteString: String?) {
+            fun writeToFile(bufferedWriter: BufferedWriter) {
+                with(bufferedWriter) {
+                    write(noteString)
+                    close()
+                }
+            }
+            if (path != null) {
+                val file = path.toFile()
+                val noteFile = File(file, "$DUMMY_FILENAME.$NOTE_EXT")
+                if (!noteFile.exists()) {
+                    try {
+                        val fileWriter = FileWriter(noteFile)
+                        val bufferedWriter = BufferedWriter(fileWriter)
+                        writeToFile(bufferedWriter)
+
+                        val id = computeId(Files.size(noteFile.toPath()), noteFile.toPath())
+
+                        Log.d("Filename", noteFile.name)
+
+                        with(noteFile){
+                            val newFile = File(file, "$id.$NOTE_EXT")
+
+                            if(!newFile.exists())
+                                if (renameTo(newFile))
+                                    Log.d("New filename", newFile.name)
+                            else
+                                removeFileFromMemory(noteFile.toPath())
+
+                        }
+
+                    } catch (e: Exception) {
+                        e.printStackTrace()
+                    }
+                }
+            }
+        }
+
+        private fun removeFileFromMemory(path: Path?) {
+            if (path != null)
+                Files.deleteIfExists(path)
+        }
+
+        private fun countNotes(path: Path?): Int{
+            var numberOfNotes = 0
+            if(path != null)
+                Files.list(path).forEach {
```

Comment:
Should possible to just take `length` or `size` without new variable and manual traversal.

Reply:
Or `count()`: https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#count--

Reply:
I don't know if we need that function, I was using it for debugging, just forgot to remove it on commit. Is it really important?

Reply:
@ShubertMunthali if it's not needed for production, let's just remove it.